### ISSUE TITLE
Bridge does not transfer retained messages at startup (Reopening #701)

### ISSUE THREAD

**Issue by sirockin** [eclipse/mosquitto#1084]
(Duplicates #701 but I don't have the privilleges to repopen)
Running from Docker Image `eclipse-mosquitto:1.5.4` (same results for 1.5.3)

Test procedure:
- Start both brokers (from docker-compose.yml, below)
- Subscribe to '#' on both brokers
- Publish a retained message "hello" to topic "foo" on broker1 - message received from both brokers
- Disconnect both subscriptions and resubscribe to confirm retained - message received from both brokers
- Restart broker 2 (`docker-compose stop broker2`,  `docker-compose start broker2`)
- Resubscribe to broker 2 '#' - no messages received

docker-compose.yml:
```
version: '3' # docker compose version
services:
  broker1:
    image: eclipse-mosquitto:1.5.4
    restart: always
    expose:
      - 1883
    ports:
      - "1883:1883"
    volumes:
      - ./mosquitto/broker1/:/mosquitto/
  broker2:
    image: eclipse-mosquitto:1.5.4
    restart: always
    expose:
      - 1883
    ports:
      - "1884:1883"
    volumes:
      - ./mosquitto/broker2/:/mosquitto/

```

broker1 config:
```
persistence false
listener 1883
log_type all

connection bridge
remote_clientid broker1
topic # out 0 "" ""
restart_timeout 5
address broker2:1883
```
broker2 config:
```
persistence false
listener 1883
log_type all
```

**Comment by karlp** [comment]
It's on the changelog for the 1.5.5 release, so I'd say it just missed getting into 1.5.3 as the comments there imply.  You're testing with 1.5.4

**Comment by sirockin** [comment]
Thanks @karlp - yes the issue suggested a fix in 1.5.3 and I also thought I'd checked that 1.5.4 was latest. 
I've just pulled 1.5.5 and it's now working as designed.
